onadata/apps/api/viewsets/open_data_viewset.py

```diff
@@ -53,12 +54,51 @@ def process_tableau_data(data, xform):
     with the column header fields for the same form.
     Handles Flattenning repeat data for tableau
     """
+
+    def get_xpath(key, nested_key):
+        index_tags = DEFAULT_INDEX_TAGS
+        val = nested_key.split('/')
+        nested_key_diff = val[len(key.split('/')):]
+        xpaths = [
+            '{key}{open_tag}{index}{close_tag}'.format(
+                key=key if len(key.split('/')) > 1
+                else nested_key.split('/')[0],
+                open_tag=index_tags[0],
+                index=index,
+                close_tag=index_tags[1])
+                ]
+        if len(key.split('/')) > 1:
+            try:
+                xpaths = xpaths[0] + '/' + nested_key_diff
+            except TypeError:
+                xpaths = xpaths[0] + ('/').join(nested_key_diff)
+        else:
+            xpaths = xpaths + [nested_key.split('/')[1]]
```

Comment:
SInce we've referenced `nested_key.split('/')` in line 76 and 60, may be we want to set the call in a variable then call the variable on those lines.

Reply:
Alright, let me update this